app/reducers/channels.js

```diff
@@ -44,19 +43,113 @@ export const RECEIVE_SUGGESTED_NODES_ERROR = 'RECEIVE_SUGGESTED_NODES_ERROR'
 export const RECEIVE_SUGGESTED_NODES = 'RECEIVE_SUGGESTED_NODES'
 
 // ------------------------------------
-// Actions
+// Helpers
 // ------------------------------------
-export function setChannelForm(form) {
-  return {
-    type: SET_CHANNEL_FORM,
-    form
+
+/**
+ * Get a name to display for the channe.
+ *
+ * This will either be:
+ *  - the alias of the node at the othe end of the channel
+ *  - a shortened public key.
+ * @param  {Object} channel Channel object.
+ * @param  {Array} nodes Array of nodes.
+ * @return {SAtring} Channel display name.
+ */
+const getDisplayName = (channel, nodes) => {
+  const remoteNodePubkey = getRemoteNodePubKey(channel)
+  const node = nodes.find(n => n.pub_key === remoteNodePubkey)
+
+  if (node && node.alias && node.alias.length) {
+    return node.alias
+  }
+
+  return remoteNodePubkey.substring(0, 10)
+}
+
+/**
+ * Get the remote pubkey depending on what type of channel.
+ *
+ * due to inconsistent API vals the remote nodes pubkey will be under remote_pubkey for active channels and
+ * remote_node_pub for pending channels.
+ * we have
+ * @param  {[type]} channel [description]
+ * @return {[type]}         [description]
+ */
+const getRemoteNodePubKey = channel => {
+  return channel.remote_pubkey || channel.remote_node_pub
+}
+
+/**
+ * Determine the status of a channel.
+ * @param  {Object} channel Channel object.
+ * @param  {Array} closingChannelIds List of channel ids that we are in the process of closing.
+ * @return {String} Channel status name.
+ */
+const getStatus = (channel, closingChannelIds) => {
+  // if the channel has a confirmation_height property that means it's pending
+  if ('confirmation_height' in channel) {
+    return 'pending'
+  }
+  // if the channel has a closing tx that means it's closing
+  if ('closing_txid' in channel) {
+    return 'closing'
+  }
+  // if the channel is in waiting_close_channels phase
+  if ('limbo_balance' in channel) {
+    return 'closing'
+  }
+  // if we are in the process of closing this channel
+  if (closingChannelIds.includes(channel.chan_id)) {
+    return 'closing'
+  }
+  // if the channel isn't active that means the remote peer isn't online
+  if (!channel.active) {
+    return 'offline'
+  }
+  // if all of the above conditionals fail we can assume the node is online :)
+  return 'online'
+}
+
+/**
+ * Decorate a channbel object with additiopnal calculated properties.
+ * @param  {Object} channelObj Channel object.
+ * @param  {Array} nodes Array of node data.
+ * @param  {Array} closingChannelIds List of channel ids that we are in the process of closing.
+ * @return {Object} Decorated channel object.
+ */
+const decorateChannel = (channelObj, nodes, closingChannelIds) => {
+  // If this is a pending channel, the channel data will be stored under the `channel` key.
+  const channelData = channelObj.channel || channelObj
+
+  const display_pubkey = getRemoteNodePubKey(channelData)
```

Comment:
Can we use camelCase here please? And then name them accordingly in `updatedChannelData`

Reply:
Updated